### ISSUE TITLE
cloudflare_list cloudflare_list_item: Error message for failed importing is a bit missleading

### ISSUE THREAD

**Issue by ztNIE** [cloudflare/terraform-provider-cloudflare#3023]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

terraform -v
Terraform v1.6.6
on darwin_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v4.20.0

### Affected resource(s)

cloudflare_list
cloudflare_list_item

### Terraform configuration files

```hcl
resource "cloudflare_list" "example_list" {
    name = "example"
    account_id = var.account_id
    kind = "redirect"
}

resource "cloudflare_list_item" "example_list_item" {
    account_id = var.account_id
    list_id = "cxxxxxxxxxxxxxxxxxxxxxxxx4ad6e26"
}
```

### Link to debug output

https://gist.github.com/ztNIE/ae5456c503b66e544becb7ee0ffb7c05

### Expected output

It should import the resource successfully.

### Actual output

╷
│ Error: The provider returned a resource missing an identifier during ImportResourceState. This is generally a bug in the resource implementation for import. Resource import code should not call d.SetId("") or create an empty ResourceData. If the resource is missing, instead return an error. Please report this to the provider developers.
│ 
│ 
╵

### Steps to reproduce

1. Create a bulk redirect list in cloudflare
2. Get the id of cloudflare redirect list id and list item id through cloudflare API. ref https://developers.cloudflare.com/api/operations/lists-get-a-list, https://developers.cloudflare.com/api/operations/lists-get-list-items
3. Use the id of list and list item to try import the resource to your terraform config https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/list_item

**Comment by ztNIE** [comment]
I'm keen to work on this issue

**Comment by jacobbednarz** [comment]
this doesn't look like a bug but user error with the wrong ID?

```
filters.api.not_found: could not find list cc491d9e7bea493fa3a056ab83e03fb9
```

**Comment by ztNIE** [comment]
thanks for your timely reply jacob, I just confirmed that it's because of the wrong list ID.
Perhaps the error message can be improved tho?

**Comment by jacobbednarz** [comment]
in most cases, what is happening here is normal behaviour. the end user provides the `Import` string, we trust it and [set the identifier](https://github.com/cloudflare/terraform-provider-cloudflare/blob/7c5bbad741530bf65f16dbcf5385f540505b551b/internal/sdkv2provider/resource_cloudflare_list.go#L71), client makes a call to the API, the response is a HTTP 404 so we assume that it has been removed and [should be cleaned up from state](https://github.com/cloudflare/terraform-provider-cloudflare/blob/7c5bbad741530bf65f16dbcf5385f540505b551b/internal/sdkv2provider/resource_cloudflare_list.go#L92-L98). the tricky bit is working out whether we can trust the identifier from the end user 🤔

**Comment by ztNIE** [comment]
Yeah it's very tricky I agree and an extra API call probably doesn't worth it.

How about just make the error message provides a little bit more context?

I only created this issue because the error message said that "This is generally a bug in the resource implementation for import. ... Please report this to the provider developers." Since this error is expected when a wrong identifier is provided, could this error message contains something like "This could be caused by a wrong resource id is provided." I could fix the import command in seconds if I knew it's the id problem from the error message.

**Comment by jacobbednarz** [comment]
the `cloudflare_account_member` resource gives you a nicer message because it takes the route of making an additional API call _before_ setting the identifier.

https://github.com/cloudflare/terraform-provider-cloudflare/blob/7c5bbad741530bf65f16dbcf5385f540505b551b/internal/sdkv2provider/resource_cloudflare_account_member.go#L137-L140

while this works, it basically makes you have an additional API call per import operation (which in some cases, is doubling the API calls made).

**Comment by xav-ie** [comment]
I am wondering if maybe we encounter errors, we can turn on a debug mode somehow? I am encountering this issue with importing worker script and was very worried that something was wrong on provider side.

**Comment by jacobbednarz** [comment]
terraform has the ability through logging to view the internals - https://developer.hashicorp.com/terraform/internals/debugging

**Comment by jacobbednarz** [comment]
#4228 doesn't change anything here (despite it being ported to the framework) since we still rely on the end user to provide the correct ID.

**Comment by jhutchings1** [comment]
We're not currently focused on v4 issues, so I'm going to mark this one as a won't fix. Thanks!
